docs/ABI/TypeMetadata.rst

```diff
@@ -123,18 +123,20 @@ contain the following fields:
 
 - The function flags are stored at **offset 1**, information contained by flags
   includes 'throws' indicator (8 bits), metadata convention (8 bits)
-  and number of arguments (remaining N bits).
-- The **argument vector** begins at **offset 2** and consists of pointers to
-  metadata records of the function's arguments.
-- A reference to the **result type** metadata record is stored after the end
-  of **argument vector**. If the function has multiple returns, this references a
-  `tuple metadata`_ record.
-
-  A pointer to each argument's metadata record will be appended separately,
-  the lowest bit being set if it is **inout**. Because of pointer alignment,
-  the lowest bit will always be free to hold this tag.
-
-  If the function takes no arguments, **argument vector** is going to be empty.
+  and number of arguments (remaining 16 bits).
+- A reference to the **result type** metadata record is stored after function
+  flags. If the function has multiple returns, this references a `tuple metadata`_
+  record.
+- The **parameter vector** begins at **offset 3** and consists of pointers to
+  metadata records of the function's parameters.
```

Comment:
I don't think we need to say anything about "offset 3" here, but we should say something like:

> The **parameter type vector** follows the result type and consists of NumParameters type metadata pointers corresponding to the types of the parameters.